spring-boot/admin/src/main/java/eu/coatrack/admin/model/repository/MetricsAggregationCustomRepository.java

```diff
@@ -379,6 +376,38 @@ public long getNumberOfApiCallers(LocalDate from, LocalDate until, String apiPro
         return getSummarizedMetrics(serviceId, null, fromDate, untilDate);
     }
 
+    public List<StatisticsPerService> getStatisticsOfServicesOwnedByLoggedInUserAndByDateRange(
+            List<ServiceApi> servicesOwnedByLoggedInUser,
+            LocalDate fromDate,
+            LocalDate untilDate) {
+
+        List<StatisticsPerService> listOfStatisticsPerServiceOwnedByLoggedInUser = new ArrayList<>();
+        return addServiceStatisticToListOfStatistics(listOfStatisticsPerServiceOwnedByLoggedInUser,
+                servicesOwnedByLoggedInUser, fromDate, untilDate);
+    }
+
+    public List<StatisticsPerService> addServiceStatisticToListOfStatistics(List<StatisticsPerService> listOfStatisticsPerServiceOwnedByLoggedInUser,
+                                                                            List<ServiceApi> servicesOwnedByLoggedInUser, LocalDate fromDate,
+                                                                            LocalDate untilDate) {
+        servicesOwnedByLoggedInUser.forEach((service) -> {
+            List<MetricsAggregation> getSummarizedMetricsFilteredByMetricTypeResponse = getSummarizedMetrics(service
+                    .getId(), null, fromDate, untilDate)
+                    .stream()
+                    .filter(metric -> metric.getType() == MetricType.RESPONSE)
+                    .collect(Collectors.toList());
+
+            long totalNoOfCallsPerService = 0L;
```

Comment:
I think the code can be shortened to the following:
```java
long totalNoOfCallsPerService = 0L;
for (MetricsAggregation serviceMetrics : getSummarizedMetricsFilteredByMetricTypeResponse) {
    totalNoOfCallsPerService += serviceMetrics.getCount();
}
StatisticsPerService statisticsPerEachService = new StatisticsPerService(service.getName(), totalNoOfCallsPerService, 0);
listOfStatisticsPerServiceOwnedByLoggedInUser.add(statisticsPerEachService);
```
Maybe you could also replace the for loop by a stream using `forEach()`.

Reply:
Regarding the suggestions, I did:
- Created the subfunction `calculateTotalNumberOfCallsPerService()`, as a function/method should have max (20 lines), as we talked before;
- Replaced the "No" on the `totalNoOfCallsPerService` to "Number", so no it is `totalNumberOfCallsPerService`

Regarding the integration of the forEach(), I think that would create more complexity and also would made the method/function hard to read

Reply:
I tried with the ForEach, and I needed like 3 lines, so changing the for to a forEach is doesn't improve readability.
Based on this, I consider this thread resolved